src/azure-cli-core/azure/cli/core/commands/template_create.py

```diff
@@ -110,3 +122,132 @@ def handle_folding(namespace):
+            namespace=property_type.split('/')[0],
+            type=property_type.split('/')[1],
+            subscription=get_subscription_id())
+        value_supplied_was_id = False
+    return (resource_id_parts, value_supplied_was_id)
+
+def get_folded_parameter_validator( # pylint: disable=too-many-arguments
+        property_name, property_type, property_option,
+        parent_name=None, parent_type=None, parent_option=None,
+        allow_none=False, allow_new=False, default_none=False):
+
+    # Ensure that all parent parameters are specified if any are
+    parent_params = [parent_name, parent_type, parent_option]
+    has_parent = any(parent_params)
+    if has_parent and not all(parent_params):
+        raise CLIError('All parent parameters must be specified (name, type, option) if any are.')
+
+    if default_none and not allow_none:
+        raise CLIError('Cannot use default_none=True if allow_none=False')
+
+    # construct the validator
+    def validator(namespace):
+        type_field_name = '{}_type'.format(property_name)
+        property_val = getattr(namespace, property_name, None)
+        parent_val = getattr(namespace, parent_name, None) if parent_name else None
+
+        # Check for the different scenarios (order matters)
+        # 1) provided value indicates None (pair of empty quotes)
+        if property_val in ('', '""', "''") or (property_val is None and default_none):
+            if not allow_none:
+                raise CLIError('{} cannot be None.'.format(property_option))
+            setattr(namespace, type_field_name, 'none')
+            setattr(namespace, property_name, None)
+            if parent_name and parent_val:
+                logger = _logging.get_az_logger(__name__)
+                logger.warning('Ignoring: %s %s', parent_option, parent_val)
+                setattr(namespace, parent_name, None)
+            return # SUCCESS
+
+        # Create a resource ID we can check for existence.
+        (resource_id_parts, value_was_id) = _validate_name_or_id(
+            namespace.resource_group_name, property_val, property_type, parent_val, parent_type)
+
+        # 2) resource exists
+        if resource_exists(**resource_id_parts):
+            setattr(namespace, type_field_name, 'existingId')
+            setattr(namespace, property_name, resource_id(**resource_id_parts))
+            if parent_val:
+                if value_was_id:
+                    logger = _logging.get_az_logger(__name__)
+                    logger.warning('Ignoring: %s %s', parent_option, parent_val)
+                setattr(namespace, parent_name, None)
+            return # SUCCESS
+
+        # if a parent name was required but not specified, raise a usage error
+        if has_parent and not value_was_id and not parent_val and not allow_new:
+            raise ValueError('incorrect usage: {} ID | {} NAME {} NAME'.format(
+                property_option, property_option, parent_option))
```

Comment:
you can use `'incorrect usage: {0} ID | {0} NAME ...` to save an occurrence of `property_option`